core/src/main/scala/caliban/GraphQL.scala

```diff
@@ -100,36 +100,34 @@ trait GraphQL[-R] { self =>
             _             <- Validator.validate(document, typeToValidate)
           } yield ()
 
-        override def executeRequest(
-          request: GraphQLRequest,
-          skipValidation: Boolean,
-          enableIntrospection: Boolean,
-          queryExecution: QueryExecution
-        )(implicit trace: Trace): URIO[R, GraphQLResponse[CalibanError]] =
+        override def executeRequest(request: GraphQLRequest)(implicit
+          trace: Trace
+        ): URIO[R, GraphQLResponse[CalibanError]] =
           decompose(wrappers).flatMap {
             case (overallWrappers, parsingWrappers, validationWrappers, executionWrappers, fieldWrappers, _) =>
               wrap((request: GraphQLRequest) =>
                 (for {
                   doc              <- wrap(Parser.parseQuery)(parsingWrappers, request.query.getOrElse(""))
                   intro             = doc.isIntrospection
-                  _                <- ZIO.when(intro && !enableIntrospection) {
+                  config           <- Configurator.configuration
```

Comment:
This feels a bit dangerous as we could accidentally use it further down in the pipeline after the underlying FiberRef has been modified. I'm not sure whether there's anything we could do about it though - I guess as long as we're aware of it should be fine

Reply:
Yeah I was careful to call it again in the validation and execution code since it can be modified by a wrapper.
I agree with you so I did this trick:
```scala
config                               <- Configurator.configuration.map { config =>
                                          (config.skipValidation, config.enableIntrospection)
                                        }
(skipValidation, enableIntrospection) = config
```